sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -221,62 +222,70 @@ internal class RadarVerificationManager(
+        verificationManager.trackVerified(beacons, object : Radar.RadarTrackVerifiedCallback {
+            override fun onComplete(
+                status: Radar.RadarStatus,
+                token: RadarVerifiedLocationToken?
+            ) {
+                var expiresIn = 0
+                var minInterval: Int = interval
 
-                    token?.let {
-                        expiresIn = it.expiresIn
+                token?.let {
+                    expiresIn = it.expiresIn
 
-                        // if expiresIn is shorter than interval, override interval
-                        minInterval = minOf(it.expiresIn, interval)
-                    }
+                    // if expiresIn is shorter than interval, override interval
+                    minInterval = minOf(it.expiresIn, interval)
+                }
 
-                    // re-request early to maximize the likelihood that a cached token is available
-                    if (minInterval > 20) {
-                        minInterval -= 10
-                    }
+                // re-request early to maximize the likelihood that a cached token is available
+                if (minInterval > 20) {
+                    minInterval -= 10
+                }
 
-                    // min interval is 10 seconds
-                    if (minInterval < 10) {
-                        minInterval = 10;
-                    }
+                // min interval is 10 seconds
+                if (minInterval < 10) {
+                    minInterval = 10;
+                }
 
-                    if (verificationManager.scheduled) {
-                        verificationManager.logger.d("Token request already scheduled")
+                if (verificationManager.scheduled) {
+                    verificationManager.logger.d("Token request already scheduled")
 
-                        return
-                    }
+                    return
+                }
 
-                    verificationManager.logger.d("Requesting token again in $minInterval seconds | minInterval = $minInterval; expiresIn = $expiresIn; interval = $interval")
+                verificationManager.logger.d("Requesting token again in $minInterval seconds | minInterval = $minInterval; expiresIn = $expiresIn; interval = $interval")
 
-                    verificationManager.scheduled = true
+                verificationManager.scheduled = true
 
-                    handler.postDelayed(object : Runnable {
-                        override fun run() {
-                            if (verificationManager.started) {
-                                verificationManager.logger.d("Token request interval fired")
+                if (runnable == null) {
+                    runnable = Runnable {
+                        if (verificationManager.started) {
+                            verificationManager.logger.d("Token request interval fired")
 
-                                trackVerified()
```

Comment:
This was calling `RadarVerificationManager.trackVerified()` instead of recursively calling the lambda